### PR TITLE
Implemented the `race condition`

### PR DESCRIPTION
**Pull request by Likh-Alex**

### CODE REVIEW

**Review by vlykhvar — APPROVED** [review]

File: log/log.log [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+ 21:01:25,070 INFO  CustomExtendedThread:13 - Extended 194
+ 21:01:25,070 INFO  CustomRunnableThread:14 - Runnable 195
+ 21:01:25,070 INFO  CustomExtendedThread:13 - Extended 196
+ 21:01:25,070 INFO  CustomRunnableThread:14 - Runnable 197
+ 21:01:25,070 INFO  CustomExtendedThread:13 - Extended 198
+ 21:01:25,070 INFO  CustomRunnableThread:14 - Runnable 199
+ 21:01:25,070 INFO  CustomExtendedThread:13 - Extended 200
+ 21:01:25,071 INFO  CustomExtendedThread:13 - Extended 201
+ 21:01:25,071 INFO  CustomExtendedThread:13 - Extended 202
+
```
Comment (RomanDubovskyi) [inline log/log.log:101]: empty line at the end

File: src/main/java/CustomExtendedThread.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+public class CustomExtendedThread extends Thread {
+    private static final Logger logger = Logger.getLogger(CustomExtendedThread.class);
+    private final Counter counter;
+
+    public CustomExtendedThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    public void run() {
+        for (int i = 0; i <= 100; i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/CustomExtendedThread.java]:
```suggestion
        while (counter.getCount() < 100) {
```
you shouldn't perform this operation 100 times, just increment it till certain value

File: src/main/java/CustomRunnableThread.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    private static final Logger logger = Logger.getLogger(CustomRunnableThread.class);
+    private final Counter counter;
+
+    public CustomRunnableThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        for (int i = 0; i <= 100; i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/CustomRunnableThread.java]: same here

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: log/log.log (hunk 1 above)
Comment (Likh-Alex) [reply log/log.log:101]: Done

File: src/main/java/CustomExtendedThread.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/CustomExtendedThread.java]: refactored to use with `while` loop

File: src/main/java/CustomRunnableThread.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/CustomRunnableThread.java]: Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**